x-pack/platform/plugins/shared/stack_alerts/public/rule_types/es_query/expression/expression.tsx

```diff
@@ -60,10 +60,19 @@ export const EsQueryRuleTypeExpression: React.FunctionComponent<
     }
   );
 
-  const errorParam = ALL_EXPRESSION_ERROR_KEYS.find((errorKey) => {
-    // @ts-expect-error upgrade typescript v5.1.6
-    return errors[errorKey]?.length >= 1 && ruleParams[errorKey] !== undefined;
-  });
+  const errorParam =
+    ALL_EXPRESSION_ERROR_KEYS.find((errorKey) => {
+      return (
+        // @ts-expect-error upgrade typescript v5.1.6
+        errors[errorKey]?.length >= 1 && ruleParams[errorKey] !== undefined
+      );
+    }) ??
+    // For search source alerts, if the only error is timeField, show this error even if the param is undefined
+    // timeField is inherently a part of the selectable data view, so if the user selects a data view with no
+    // timeField, this data view is incompatible with the rule.
```

Comment:
Discover has a test that relies on this error being shown correctly. It seems like it was relying on a bug in the v1 rule form to actually work.